EpikCord/ext/tasks.py

```diff
@@ -0,0 +1,102 @@
+import datetime
+from typing import Callable, Optional
+from time import time as _time
+import asyncio
+from ..exceptions import TaskFailedError
+
+_EPOCH = datetime.datetime(1970, 1, 1)
+
+
+class TimeParser:
+    @staticmethod
+    def get_time_since_epoch(
+        year: int, month: int, day: int, hour: int, minute: int, seconds: float
+    ) -> float:
+        date = datetime.datetime(year, month, day, hour, minute, seconds)
+
+        delta = date - _EPOCH
+        return delta.total_seconds()
+
+    @staticmethod
+    def total_seconds(
+        year: int = 0,
+        month: int = 0,
+        day: int = 0,
+        hour: int = 0,
+        minute: int = 0,
+        seconds: float = 0,
+    ) -> float:
+        month += year * 365
+        day += month * 30
+        hour += day * 24
+        minute += hour * 60
+        seconds += minute * 60
+        return seconds
+
+
+class Tasks:
+    def __init__(self, client) -> None:
+        self.client = client
+
+    def add_task(
+        self,
+        task: Callable[..., None],
+        interval: Optional[int] = None,
+        start: Optional[float] = None,
+        until: Optional[float] = None,
+        **kwargs,
+    ):
+        """Adds a background task (Tasks that run silently in the background)
+
+        This task can be used to monitor the bot's performance, change and send stuff routinely
+        through webhooks, etc
+        You can set the time of when it starts, stops and  add which arguments to give to the task
+        You may also define whether to run once or run indefinitely
+
+        Args:
+            task (Callable): The function to run on the background
+            interval (Optional[int]): The interval (in seconds) for a delay between when the instance of the task stops and starts again.Defaults to 5.
+            You may use `TimeParser.total_seconds` method
+            instances (Optional[int]): The number of instances to start after delay. Usually unlimited
+            start (Optional[float]): The time to start, you may use `TimeParser.get_time_since_epoch` method to give the seconds since python epoch to this arg
+            until (optional[float]): The time to end, you may use `TimeParser.get_time_since_epoch` method to give the seconds since python epoch to this arg
+
+
+        """
+        kwargs["interval"] = interval
+        kwargs["start"] = start
+        kwargs["until"] = until
+
+        async def full_task(client, task, **kwargs):
+            async def task_func(client, interval):
```

Comment:
Since the `client` obj is accessed from outer scope, what the point of passing it again to `task_func`?

Reply:
will change that in a new commit